src/StreamMetrics.js

```diff
@@ -0,0 +1,282 @@
+                    const lastMessages = await this.getResend(this.legacyStreamId, 1)
+
+                    if (lastMessages.length === 0) {
+                        await this.publishReport()
+                    } else {
+                        const messages = await this.getResend(this.streamIds.minuteStreamId, this.legacyInterval)
+
+                        for (let i = 1; i < messages.length; i++) {
+                            this.report.broker.messagesToNetworkPerSec += messages[i].broker.messagesToNetworkPerSec
+                            this.report.broker.bytesToNetworkPerSec += messages[i].broker.bytesToNetworkPerSec
+                            this.report.network.avgLatencyMs += messages[i].network.avgLatencyMs
+
+                            this.report.broker.messagesToNetworkPerSec += messages[i].broker.messagesToNetworkPerSec
+                            this.report.broker.bytesToNetworkPerSec += messages[i].broker.bytesToNetworkPerSec
+
+                            this.report.network.avgLatencyMs += messages[i].network.avgLatencyMs
+                            this.report.network.bytesToPeersPerSec += messages[i].network.bytesToPeersPerSec
+                            this.report.network.bytesFromPeersPerSec += messages[i].network.bytesFromPeersPerSec
+                            this.report.network.connections += messages[i].network.connections
+
+                            if (metricsReport.metrics['broker/cassandra']) {
+                                this.report.storage.bytesWrittenPerSec += messages[i].storage.bytesWrittenPerSec
+                                this.report.storage.bytesReadPerSec += messages[i].storage.bytesReadPerSec
+                            }
+                        }
+
+                        if (messages.length > 0) {
+                            this.report.broker.messagesToNetworkPerSec /= messages.length
+                            this.report.broker.bytesToNetworkPerSec /= messages.length
+                            this.report.network.avgLatencyMs /= messages.length
+
+                            this.report.broker.messagesToNetworkPerSec /= messages.length
+                            this.report.broker.bytesToNetworkPerSec /= messages.length
+
+                            this.report.network.avgLatencyMs /= messages.length
+                            this.report.network.bytesToPeersPerSec /= messages.length
+                            this.report.network.bytesFromPeersPerSec /= messages.length
+                            this.report.network.connections /= messages.length
+
+                            if (metricsReport.metrics['broker/cassandra']) {
+                                this.report.storage.bytesWrittenPerSec /= messages.length
+                                this.report.storage.bytesReadPerSec /= messages.length
+                            }
+
+                            if ((lastMessages[0].timestamp + this.reportMiliseconds - now) < 0) {
```

Comment:
We shouldn't really have to calculate anything above if the condition does not hold. So this condition should be probably moved up to include all the calculations?

Reply:
Hmm you're now using the timestamp of the first message in the `sourceStream`. Wondering whether this works out alright, or would it be better to actually use the timestamp of the last message in `targetStream` to check if is time to publish?